### PR TITLE
feat(actions): Create `aws-auth` composite action

### PR DESCRIPTION
**Pull request by nafisat2**
This creates an action to authenticate to AWS using cognito

### CODE REVIEW

File: actions/aws-auth/README.md
Comment (iainlane) [inline actions/aws-auth/README.md:1]: This one can be a followup, just want to note it down here so we don't forget.

It'd be good to include an example of how to set up the role, showing a simple example of trust and permission policies, including the use of `aws:RequestTag` to show people how to make use of the claims. That would work nicely if there was a second `step` in the example which used the policy we give as an example.

**Review by iainlane — COMMENTED** [review]
thanks a lot, this is great stuff 👍 

I went for a round of comments aiming to hide some of the complexity of the underlying implementation which I _think_ we can get away with not showing to the users.

Let me know what you think!

File: actions/aws-auth/README.md
Comment (nafisat2) [reply actions/aws-auth/README.md:1]: Will include in a subsequent PR 👍

File: actions/aws-auth/action.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    value: ${{ steps.auth.outputs.aws_access_key_id }}
+  aws_secret_access_key:
+    description: "AWS Secret Access Key"
+    value: ${{ steps.auth.outputs.aws_secret_access_key }}
+  aws_session_token:
+    description: "AWS Session Name"
+    value: ${{ steps.auth.outputs.aws_session_token }}
+  aws_region:
+    description: "AWS Region"
+    value: ${{ steps.auth.outputs.aws_region }}
```
[the remark this answers is not in the document]
Comment (nafisat2) [reply actions/aws-auth/action.yaml]: Added `aws-region` input

File: actions/aws-auth/action.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    required: true
+    description: "Claims from ID token to pass to the role when performing role chaining. Can also include `repository_name`, which will be calculated."
+  chain-set-in-environment:
+    default: true
+    required: false
+    description: "Set role chained credentials as environment variables"
+  chain-role-duration-seconds:
+    default: "3600"
+    required: false
+    description: "Role duration in seconds for chained role"
```
[the remark this answers is not in the document]
Comment (nafisat2) [reply actions/aws-auth/action.yaml]: Done!

File: actions/aws-auth/action.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+inputs:
+  chain-role-arn:
+    default: ""
+    required: true
+    description: "Arn of Role when performing role chaining"
+  chain-pass-claims:
+    default: "repository_owner, repository_name, job_workflow_ref"
+    required: true
+    description: "Claims from ID token to pass to the role when performing role chaining. Can also include `repository_name`, which will be calculated."
```
[the remark this answers is not in the document]
Comment (nafisat2) [reply actions/aws-auth/action.yaml]: Updated descriptions in both

**Review by iainlane — APPROVED** [review]
looks great! 🚀🌕

**Pull request merged**